docs/technical-overview.md

```diff
@@ -0,0 +1,227 @@
+
+git is a fully-featured software project, and as such we will not cover it here. If you'd like to learn more about git and how to use it, you might find the following resources helpful.
+
+For git itself:
+
+[https://git-scm.com/]()
+
+[https://git-scm.com/book/en/v2]()
+
+For using GitHub:
+
+[https://help.github.com/categories/bootcamp/]()
+
+### Package Managers
+
+When developing software, it is often useful to be able to re-use software that others have already written. Software written for the purpose of being used to help develop other software is usually referred to as a *library*. This project uses several software libraries, which we call *dependencies* of the project. We call them dependencies since our project depends on these libraries being installed on the system in order to function. Sometimes, software libraries have dependencies themselves, and then it is possible for these dependencies to have further dependencies, and so on.
+
+Manually managing the dependencies of a software package is tedious, and can also be error-prone. Consider the case where we are using a library which is pinned to a specific version of another library. We would first have to know that this dependency existed, and then have to make sure the exact version of the dependency was installed on our system. Now suppose we were using a second library which had the same dependency, but was pinned to a different version of the software. In this case, we would have to make sure that we had *both* versions of the dependency installed, and further, we would need to make sure that each library we were using actually linked to the correct version of the dependency at build-time.
+
+To manage this complexity, we use *package managers*. Software libraries are *packaged* when they are released, and these *packages* keep track of the dependencies needed for the library, which are in turn packages themselves. When we install a package with a package manager, it checks if the dependencies for that package are already installed on the system. If not, it installs their dependencies, which might involve installing the dependencies for thos packages, and so on. The package also describes which versions of the dependency are valid to use with the package, and the package manager will handle having multiple versions of the same software installed on the system.
+
+In this project we use several package managers. We will give a brief overview of each of them.
+
+#### pip
+
+[pip](https://pip.pypa.io/en/stable/) is a package manager for installing software packages written in Python. Many of the packages installed by pip can be found in the [Python Package Index](https://pypi.python.org/pypi/) also referred to as PyPi.
+
+##### Installation
+
+[https://packaging.python.org/installing/#install-pip-setuptools-and-wheel]()
+
+##### Usage
+
```

Comment:
Sure, I can understand that. I guess what I'm wondering is if should answer the question: "Why are these tools useful and how can I use them with this library?". Maybe the question of how to use them more generally should be left to man and web pages, and we can just provide pointers to them. Ultimately, we have to choose which examples to show and not show, so I think showing those that have a greater connection to this library in particular will be more impactful. And it will make it more likely that more students will want to read them.